### PR TITLE
Feature/store user infos in firestore

### PR DESCRIPTION
**Pull request by Lyokolux**
## Description

Implement POST request to firestore

## Checklist

- [x] I compare my branch with `develop`
- [x] All tests passed after running `yarn s-check && yarn lint`

### CODE REVIEW

File: src/pages/UserInfos/UserInfos.svelte [hunk 1]
```diff
@@ -30,8 +31,16 @@
...
+      const payload = {
+        age: userInfos.age,
+        sexe: userInfos.sexe,
+        anyExperience: userInfos.anyExperience,
+        experienceGrades: userInfos.experienceGrades.map((experienceGrade) => ({
+          labels: experienceGrade.labels,
+          grade: experienceGrade.grade
+        }))
+      }
+      api.addUserRequest(payload)
```
Comment (johannchopin) [inline src/pages/UserInfos/UserInfos.svelte:43]: Handle async until request done to show loading animation on button next. Maybe first do #53 and use it here.

File: src/store/api.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { initFirestore } from '../firestore'
+import {
+  collection,
+  addDoc,
+  updateDoc,
+  arrayUnion,
+  DocumentReference
+} from 'firebase/firestore'
+import { USER_COLLECTION_NAME } from '../const'
+import { writable } from 'svelte/store'
```
Comment (johannchopin) [inline src/store/api.ts:10]: Move `import` from packages to the top of the file.

File: src/store/api.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  const currentUserDoc = writable<DocumentReference>()
+
+  const addUserRequest = async (user) => {
+    currentUserDoc.set(await addDoc(dbCollection, user))
+  }
+
+  /**
+   * @param currentUserDoc provided in the store
+   */
+  const addExperimentRequest = async (userDoc, experiment) => (
```
Comment (johannchopin) [inline src/store/api.ts:25]: Please use the  TS types

**Review by johannchopin — APPROVED** [review]
Please do the fixes and then you can merge

File: src/pages/UserInfos/UserInfos.svelte (hunk 1 above)
Comment (Lyokolux) [reply src/pages/UserInfos/UserInfos.svelte:44]: Ok

File: src/store/api.ts (hunk 2 above)
Comment (Lyokolux) [reply src/store/api.ts:10]: ✅

File: src/store/api.ts (hunk 3 above)
Comment (Lyokolux) [reply src/store/api.ts]: 🎨(api.ts)

File: src/pages/UserInfos/UserInfos.svelte [hunk 4]
```diff
@@ -30,8 +31,16 @@
...
   const onSubmit = (): void => {
     schema.validate(userInfos).then(() => {
-      // TODO: Store data in firestore
-      console.log(userInfos)
+      const payload = {
+        age: userInfos.age,
+        sexe: userInfos.sexe,
+        anyExperience: userInfos.anyExperience,
+        experienceGrades: userInfos.experienceGrades.map((experienceGrade) => ({
+          labels: experienceGrade.labels,
```
[the remark this answers is not in the document]
Comment (Lyokolux) [reply src/pages/UserInfos/UserInfos.svelte]: ✅

**Pull request merged**